src/app/main.cpp

```diff
@@ -65,6 +72,35 @@ void initGraphics()
 
 int main( int argc, char **argv )
 {
+  if ( argc > 1 && strcmp( argv[1], "--service" ) == 0 )
+  {
+#if defined( Q_OS_ANDROID )
+    QAndroidService app( argc, argv );
+#else
+    QCoreApplication app( argc, argv );
+#endif
+
+    QCoreApplication::setOrganizationName( "OPENGIS.ch" );
+    QCoreApplication::setOrganizationDomain( "opengis.ch" );
+    QCoreApplication::setApplicationName( qfield::appName );
+    QSettings settings;
+
+    QEventLoop loop( &app );
+    QFieldCloudConnection connection;
+    QObject::connect( &connection, &QFieldCloudConnection::pendingAttachmentsUploadFinished, &loop, &QEventLoop::quit );
+    int pendingAttachments = connection.uploadPendingAttachments();
+    if ( pendingAttachments > 0 )
+    {
+      loop.exec();
+    }
+
+#if defined( Q_OS_ANDROID )
+    QtAndroid::androidService().callMethod<void>( "stopSelf" );
+#endif
+
+    return 0;
```

Comment:
Can we move this part to a separate file (or even better separate library)?

Reply:
@m-kuhn , I don't think we can do a separate library here, we need it to be in the same libqfield.so for reasons we discussed on google chat (i.e. static library size). Or am I misunderstanding you here?

As for a separate file, do you have a suggestion for the filename? And we'd do a static function?

Reply:
1. A new subdir "cloudsync"
2. A new CMakeLists that adds a library => will be a static library on android, so same `.so`, but _can_ be included as separate `.so` on other platforms, or even be a separate process on other platforms.
3. cloudsyncservice.cpp?